File: src/main/java/com/skidders/sigma/utils/misc/MouseHandler.java
package com.skidders.sigma.utils.misc;

import org.lwjgl.glfw.GLFW;

public class MouseHandler {

    private boolean mouseClicked = false;
    private boolean mouseButtonPressed = false;

    public MouseHandler(long windowHandle) {
        // Set up the mouse button callback
        GLFW.glfwSetMouseButtonCallback(windowHandle, (window, button, action, mods) -> {
            if (button == GLFW.GLFW_MOUSE_BUTTON_LEFT) { // Left mouse button
                if (action == GLFW.GLFW_PRESS) { // Button is pressed
                    mouseButtonPressed = true;
                } else if (action == GLFW.GLFW_RELEASE) { // Button is released
                    mouseButtonPressed = false;
                    mouseClicked = false;
                }
            }
        });
    }

    public boolean isMouseClicked() {
        if (mouseButtonPressed && !mouseClicked) {
            mouseClicked = true;
            return true; // Register click
        }
        return false; // No new click detected
    }

}

File: src/main/kotlin/com/skidders/sigma/utils/misc/MouseHandler.kt
package com.skidders.sigma.utils.misc;

import org.lwjgl.glfw.GLFW;

public class MouseHandler {

    private val windowHandle: Long;
    private var mouseClicked = false;

    constructor(windowHandle: Long) {
        this.windowHandle = windowHandle;
    }

    public fun isMouseClicked(): Boolean {
        // Check if the left mouse button is pressed
        var mouseButtonPressed = GLFW.glfwGetMouseButton(windowHandle, GLFW.GLFW_MOUSE_BUTTON_LEFT) == GLFW.GLFW_PRESS;

        if (mouseButtonPressed && !mouseClicked) {
            mouseClicked = true;
            return true; // Register click
        } else if (!mouseButtonPressed) {
            mouseClicked = false; // Reset click state when button is released
        }

        return false; // No new click detected
    }
}